systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,214 @@
+     * │                                                                                                  │
+     * │   +------------------+    +------------------+    +------------------+    +------------------+   │
+     * │   | Thread 1         |    | Thread 2         |    | Thread 3         |    | Thread N         |   │
+     * │   | +--------------+ |    | +--------------+ |    | +--------------+ |    | +--------------+ |   │
+     * │   | | Batch 1      | |    | | Batch 2      | |    | | Batch 3      | |    | | Batch N      | |   │
+     * │   | |              | |    | |              | |    | |              | |    | |              | |   │
+     * │   | | 1. Creation  | |    | | 1. Creation  | |    | | 1. Creation  | |    | | 1. Creation  | |   │
+     * │   | | 2. Modific.  | | -->| | 2. Modific.  | | -->| | 2. Modific.  | | -->| | 2. Modific.  | |   │
+     * │   | | 3. Deletion  | |    | | 3. Deletion  | |    | | 3. Deletion  | |    | | 3. Deletion  | |   │
+     * │   | +--------------+ |    | +--------------+ |    | +--------------+ |    | +--------------+ |   │
+     * │   +------------------+    +------------------+    +------------------+    +------------------+   │
+     * │                                                                                                  │
+     * │ - Each thread processes a specific batch, performing creation, modification, and deletion        │
+     * │   sequentially, but each batch operation is independent and concurrent across threads.           │
+     * │ - Threads are managed by an executor service that efficiently schedules each batch               │
+     * │   task to maximize the available CPU resources.                                                  │
+     * │ - Errors in any batch operation are logged and do not impact the processing of other             │
+     * │   batches, ensuring robust execution.                                                            │
+     * └──────────────────────────────────────────────────────────────────────────────────────────────────┘
+     *
+     * @param testStorage    The storage instance containing the configuration and namespace details.
+     * @param numberOfTopics The total number of Kafka topics to process.
+     */
+    public static void processKafkaTopicBatchesAsync(TestStorage testStorage, int numberOfTopics) {
+        final int availableCPUs = Runtime.getRuntime().availableProcessors();
+        final int optimalBatchCount = Math.max(1, availableCPUs);
+        final int topicsPerBatch = (numberOfTopics + optimalBatchCount - 1) / optimalBatchCount; // Ensures all topics are covered
+
+        final ExecutorService executor = Executors.newFixedThreadPool(optimalBatchCount);
+        final ExtensionContext currentContext = ResourceManager.getTestContext();
+        final CompletableFuture<?>[] futures = new CompletableFuture[optimalBatchCount];
+
+        try {
+            for (int batch = 0; batch < optimalBatchCount; batch++) {
```

Comment:
Should we do some warmup before this line?

Reply:
Does it really change anything?